### PR TITLE
feat: (Platform) Table Page Scrolling

### PR DESCRIPTION
**Pull request by dimamarksman**
#### Please provide a link to the associated issue.
Closes https://github.com/SAP/fundamental-ngx/issues/4344
#### Please provide a brief summary of this pull request.

This PR introduces table page scrolling feature.

The table layout has been changed significantly in order to get vertical and horizontal scrollbars.  

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

**Review by KevinOkamoto — APPROVED** [review]
Looks good.

**Comment by manjunathanagaraj** [comment]
@dimamarksman , I dont see an example for horizontal scrolling.

**Comment by dimamarksman** [comment]
> @dimamarksman ,
> 
> 1. I dont see an example for horizontal scrolling.
> 2. How does the developer control on number of rows to be displayed? I think this as well should appear in documentation, right?

In order to restrict table body height and get the vertical scroll we need to use `bodyHeight` option
image


Horizontal scrollbar appears automaticaly, there is option to enable it. This scrollbar you can see on column freezing example
image

File: libs/platform/src/lib/components/table/table.component.scss [hunk 1]
```diff
@@ -1,42 +1,206 @@
...
-    .fd-table__row--group {
+    .fdp-table__body {
+        position: relative;
+
+        overflow: hidden;
+        overflow-y: scroll;
+
+        .fd-table--fixed {
+            overflow: hidden;
+            overflow-x: scroll;
```
Comment (JKMarkowski) [inline libs/platform/src/lib/components/table/table.component.scss:151]: Is there any reason to not having `overflow-x/y  auto`?

Comment (dimamarksman) [reply libs/platform/src/lib/components/table/table.component.scss:151]: Yeah,
The trick is the parent has only vertical scroll and the child has only horizontal scroll

**Comment by puru-hk** [comment]
Testing the scrolling functionality it is working. But from the accessibility prospective not able to scroll from the keyboard. 
And I have raised some issue regarding the filter

File: libs/platform/src/lib/components/table/table.component.scss (hunk 1 above)
Comment (dimamarksman) [reply libs/platform/src/lib/components/table/table.component.scss:152]: I probaly got you wrong.
We have mock scrollbar area in the header and need to track the scrolbar width and set it there in order to have the same header width as the body width. 
So just for simplicity I added `overflow-x: scroll` in order to keep this width as a constant.
In the future we could improve it and add a proper tracking of it.

**Pull request merged**